### PR TITLE
Fix forall predicates

### PR DESCRIPTION
**Pull request by virgil-serbanuta**
---

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/src/Kore/Internal/Pattern.hs [hunk 1]
```diff
@@ -94,6 +95,11 @@ isSimplified :: Pattern variable -> Bool
 isSimplified (splitTerm -> (t, p)) =
     TermLike.isSimplified t && Predicate.isSimplified p
 
+markSimplified :: Pattern variable -> Pattern variable
```
[the remark this answers is not in the document]
Comment (virgil-serbanuta) [reply kore/src/Kore/Internal/Pattern.hs]: Oops, yes.

File: kore/src/Kore/Step/Simplification/Forall.hs [hunk 2]
```diff
@@ -98,8 +118,25 @@ makeEvaluate
 makeEvaluate variable patt
   | Pattern.isTop patt    = Pattern.top
   | Pattern.isBottom patt = Pattern.bottom
+  | not variableInTerm && not variableInPredicate = patt
+  | predicateIsBoolean =
+    TermLike.markSimplified (mkForall variable term)
```
Comment (ana-pantilie) [inline kore/src/Kore/Step/Simplification/Forall.hs]: Should we check that the `term` is simplified here? Or do we only call this when we're sure it was simplified before.

File: kore/src/Kore/Step/Simplification/Forall.hs [hunk 3]
```diff
@@ -98,8 +118,25 @@ makeEvaluate
...
+    `Conditional.withCondition` predicate
+  | termIsBoolean =
+    term
+    `Conditional.withCondition` Predicate.markSimplified
+        (Predicate.fromPredicate
+            (makeForallPredicate variable (Predicate.toPredicate predicate))
```
Comment (ana-pantilie) [inline kore/src/Kore/Step/Simplification/Forall.hs]: Same as above for the `predicate`.

File: kore/src/Kore/Step/Simplification/Forall.hs (hunk 2 above)
Comment (virgil-serbanuta) [reply kore/src/Kore/Step/Simplification/Forall.hs]: I ask @ttuegel the same question in another PR :)

I think we just assume that the input to all these functions (simplify, makeEvaluate, and so on) is already simplified.

File: kore/src/Kore/Step/Simplification/Forall.hs (hunk 3 above)
Comment (virgil-serbanuta) [reply kore/src/Kore/Step/Simplification/Forall.hs]: Same answer

**Comment by virgil-serbanuta** [comment]
> I left some questions, also I think that the docs in `kore/src/Kore/Step/Simplification/Forall.hs` from line 65 to 78 might need to be updated.

Thanks! Fixed.

File: kore/src/Kore/Step/Simplification/Forall.hs (hunk 2 above)
Comment (ana-pantilie) [reply kore/src/Kore/Step/Simplification/Forall.hs]: Ah, right. Found the conversation.

**Pull request merged**